src/main/java/me/jellysquid/mods/sodium/client/render/pipeline/FluidRenderer.java

```diff
@@ -71,8 +71,18 @@ public FluidRenderer(LightPipelineProvider lighters, BiomeColorBlender biomeColo
         this.biomeColorBlender = biomeColorBlender;
     }
 
-    private boolean isFluidExposed(BlockRenderView world, int x, int y, int z, Fluid fluid) {
-        BlockPos pos = this.scratchPos.set(x, y, z);
+    private boolean isFluidExposed(BlockRenderView world, int x, int y, int z, Direction dir, Fluid fluid) {
+        // Up direction is hard to test since it doesnt fill the block
+        if(dir != Direction.UP) {
+            BlockPos pos = this.scratchPos.set(x, y, z);
+            BlockState blockState = world.getBlockState(pos);
+            VoxelShape shape = blockState.getCullingShape(world, pos);
+            if (VoxelShapes.isSideCovered(VoxelShapes.fullCube(), shape, dir.getOpposite())) {
```

Comment:
`if (blockState.getBlock() != Blocks.KELP_PLANT && VoxelShapes.isSideCovered(VoxelShapes.fullCube(), shape, dir.getOpposite())) {`

Reply:
`if (blockState.isOpaque() && VoxelShapes.isSideCovered(VoxelShapes.fullCube(), shape, dir.getOpposite())) {` looks to be more ideal. as this should then work for any mods that add in transparent fullCube blocks.